### DISCUSSION TITLE
Handle different turbine types

### DISCUSSION THREAD

**Discussion by roberttum** [NREL/flasc#196] [Q&A]
Dear flasc developers,

first of all thanks for this nice package!

For the northing calibration based on enery ratios, I could not easily understand if having different types of turbines could be an issue. In my example, the farm has 4 different machine types (2-4 MW) at different hub heights (ca 100 turbines). This is all reflected in the FLORIS v4 model. 
Can flasc do the northing calibration through energy ratios with the cleaned SCADA data for all turbines? Or should I e.g. filter for similar turbines types / normalize all power output?

Greetings
Robert

**Comment by misi9170** [comment]
Hi @roberttum , thanks for your question! This is interesting, and I have to say, not something that we have thought about much. Speaking a bit with @ejsimley about it, we think it should be possible to use energy ratios where the reference turbine is a different model (and rated power) than the test turbine. As you mention, this would also need to be reflected in the FLORIS model when running a northing calibration so that the "unwaked" energy ratio (which is usually 1, when a reference turbine is the same as a test turbine) is correct.

To be clear, we have not tried to do this ourselves and there could be quirks that we haven't thought through. If you try this and it isn't working as you expect, we'd certainly be interested to hear about it!

**Reply by roberttum** [reply]
Thanks for your answer Misha! I will test it a bit more and let you know if something unexpected happens.

Greetings
Robert
